src/tasks/WasmBuildTasks/GetChromeVersions.cs

```diff
@@ -78,38 +82,85 @@ private async Task<bool> ExecuteInternalAsync()
         }
     }
 
-    private async Task<string> GetChromeUrlsAsync(ChromeVersionSpec version)
+    private async Task<(ChromeVersionSpec versionSpec, string baseSnapshotUrl)> FindVersionFromHistoryAsync()
     {
-        string baseUrl = $"{s_snapshotBaseUrl}/{OSPrefix}";
+        List<string> versionsTried = new();
 
-        int branchPosition = int.Parse(version.branch_base_position);
-        for (int i = 0; i < s_numBranchPositionsToTry; i++)
+        await foreach (string version in GetVersionsAsync())
```

Comment:
Should probably set a limit on how many versions we try

Reply:
Maybe it should also consider the major version, and only try like 2 previous major versions.